components/director/internal/repo/list.go

```diff
@@ -0,0 +1,48 @@
+package repo
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/pkg/errors"
+
+	"github.com/kyma-incubator/compass/components/director/internal/persistence"
+)
+
+type Lister struct {
+	tableName       string
+	selectedColumns string
+	tenantColumn    string
+}
+
+func NewLister(tableName, tenantColumn string, selectedColumns []string) *Lister {
+	return &Lister{
+		tableName:       tableName,
+		selectedColumns: strings.Join(selectedColumns, ", "),
+		tenantColumn:    tenantColumn,
+	}
+}
+
+func (l *Lister) List(ctx context.Context, tenant string, dest Collection, additionalConditions ...string) error {
+	persist, err := persistence.FromCtx(ctx)
+	if err != nil {
+		return err
+	}
+
+	filterSubquery := ""
+	for _, cond := range additionalConditions {
+		if strings.TrimSpace(cond) != "" {
+			filterSubquery += fmt.Sprintf(` AND %s`, cond)
+		}
+	}
+
+	stmt := fmt.Sprintf("SELECT %s FROM %s WHERE %s=$1 %s", l.selectedColumns, l.tableName, l.tenantColumn, filterSubquery)
```

Comment:
Can we share the way in which we creating this query, because we do in the same way in `list with pagination`.
For example: Extract to method, like we have in `common` `appendEnumeratedConditions` method